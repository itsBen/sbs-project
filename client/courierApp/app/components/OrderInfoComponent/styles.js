import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  container: {
    flex: 1,
    alignSelf: 'stretch',
  },
  footer: {
    flexDirection: 'row',
    paddingHorizontal: 22,
    paddingVertical: 10,
  },
  footerBody: {
    flex: 1,
  },
  pricelabel: {
    fontSize: 19,
    fontWeight: 'bold',
    textAlign: 'center',
    marginTop: 10,
    marginBottom: 20,
  },
  quantitySelector: {
    marginBottom: 20,
    flexDirection: 'row',
    justifyContent: 'space-around',
  },
  quantitySelector_value: {
    fontSize: 15,
    alignSelf: 'center',
  },
  metatable: {
    borderWidth: 1,
    borderColor: '#e0e0e0',
    borderRadius: 2,
    marginBottom: 20,
  },
  metatableElement: {
    borderBottomWidth: 1,
    borderBottomColor: '#e0e0e0',
    paddingTop: 5,
    paddingBottom: 5,
    paddingLeft: 20,
    paddingRight: 20,
  },
  metatableElementLabel: {
    color: '#333333',
    marginBottom: 3,
  },
  metatableElementValue: {
    color: '#4D4D4D',
  },
});
