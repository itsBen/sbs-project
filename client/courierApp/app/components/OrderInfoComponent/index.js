import React, { Component } from 'react';
import { Image, Text, View, ScrollView, Alert } from 'react-native';
import { TableView, Section, Cell } from 'react-native-tableview-simple';
import Modal from 'react-native-modal';
import * as firebase from 'firebase';
import moment from 'moment';

import styles from './styles';
import BackgroundImage from '@components/BackgroundImage';
import Spinner from '@components/Spinner';
import ProductInfoDialog from '@components/ProductInfoDialog';
import ListHeaderSection from '@components/ListHeaderSection';
import OrderDetailsCell from '@components/OrderDetailsCell';
import Button from '@components/Button';
import { getTotalPrice } from '@utilities';

export default class extends Component {
  state = {
    currentDetailsProductId: null,
  };

  constructor(props) {
    super(props);
    this.handleOpenDetails = this.handleOpenDetails.bind(this);
  }

  handleOpenDetails(productId) {
    this.setState({
      currentDetailsProductId: productId,
    });
  }

  render() {
    const { order, buttonDisabled } = this.props;
    // Render Modal every time to allow animations
    const renderProductInfoModal = () => (
      <Modal
        isVisible={!!this.state.currentDetailsProductId}
        animationIn={'slideInRight'}
        animationOut={'slideOutRight'}
      >
        <ProductInfoDialog
          productId={this.state.currentDetailsProductId}
          onClose={() => this.setState({ currentDetailsProductId: null })}
        />
      </Modal>
    );
    return (
      <BackgroundImage>
        {!order ? (
          <Spinner />
        ) : (
          <View style={styles.container}>
            {renderProductInfoModal()}
            <ScrollView>
              <TableView>
                <ListHeaderSection
                  header="Grocery List"
                  footer={'Total: ' + getTotalPrice(order.products) + ' €'}
                >
                  {Object.values(order.products).map(product => (
                    <Cell
                      key={product.id}
                      cellStyle="RightDetail"
                      title={product.name}
                      detail={' ~ ' + product.price + ' €'}
                      accessory="DisclosureIndicator"
                      onPress={() => this.handleOpenDetails(product.productId)}
                      image={
                        <Image
                          style={{ borderRadius: 5 }}
                          source={{
                            uri:
                              'https://facebook.github.io/react/img/logo_og.png',
                          }}
                        />
                      }
                    />
                  ))}
                </ListHeaderSection>

                <ListHeaderSection header="Details" sectionPaddingBottom={0}>
                  <Cell
                    cellContentView={
                      <OrderDetailsCell
                        label="Location"
                        info={'Near ' + order.location}
                        iconName="location-on"
                      />
                    }
                  />
                  <Cell
                    cellContentView={
                      <OrderDetailsCell
                        label="Deliver by"
                        info={moment(order.timeLimit).format('MM.DD.YY h:mm a')}
                        iconName="access-time"
                      />
                    }
                  />
                  <Cell
                    cellContentView={
                      <OrderDetailsCell
                        label="Earn delivery fee"
                        info={order.deliveryFee + ' €'}
                        iconName="attach-money"
                      />
                    }
                  />
                  <Cell
                    cellContentView={
                      <OrderDetailsCell
                        label="Status"
                        info={order.status}
                        iconName="info"
                      />
                    }
                  />
                  {order.purchasedAt && (
                    <Cell
                      cellContentView={
                        <OrderDetailsCell
                          label="Purchased at"
                          info={moment(order.purchasedAt).format(
                            'MM.DD.YY h:mm a'
                          )}
                          iconName="shopping-basket"
                        />
                      }
                    />
                  )}
                </ListHeaderSection>
              </TableView>
            </ScrollView>

            {!this.props.hideButton && (
              <View style={styles.footer}>
                <View style={styles.footerBody}>
                  <Button
                    title={this.props.buttonText}
                    disabled={this.props.buttonDisabled}
                    onPress={this.props.onButtonPress}
                  />
                </View>
              </View>
            )}
          </View>
        )}
      </BackgroundImage>
    );
  }
}
