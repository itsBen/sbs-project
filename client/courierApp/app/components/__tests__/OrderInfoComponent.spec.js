/* eslint-env node, jest */
import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import OrderInfoComponent from '../OrderInfoComponent';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <OrderInfoComponent
        key="34"
        order={{
          orderId: '23324',
          products: [{ name: 'Meat', price: 3.4, productId: '2342343' }],
          deliveryFee: 12,
          timeLimit: '2017-10-05T00:15:08+03:00',
          status: 'pending',
          purchasedAt: '2017-10-05T00:15:08+03:00',
        }}
        buttonDisabled={true}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
