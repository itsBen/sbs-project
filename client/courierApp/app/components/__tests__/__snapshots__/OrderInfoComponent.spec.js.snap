// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "height": null,
      "width": null,
    }
  }
>
  <Image
    source={1}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": null,
          "width": null,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignSelf": "stretch",
        "flex": 1,
      }
    }
  >
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      onModalHide={[Function]}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "backgroundColor": "black",
            "bottom": 0,
            "height": 1334,
            "left": 0,
            "opacity": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "width": 750,
          }
        }
        testID={undefined}
      />
      <View
        collapsable={undefined}
        onModalHide={[Function]}
        pointerEvents="box-none"
        style={
          Object {
            "flex": 1,
            "justifyContent": "center",
            "margin": 37.5,
            "transform": Array [
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#fff",
              "borderColor": "rgba(0, 0, 0, 0.1)",
              "borderRadius": 5,
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(230,37,101)",
                "borderTopLeftRadius": 5,
                "borderTopRightRadius": 5,
                "flexDirection": "row",
                "minHeight": 50,
                "position": "relative",
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                disabled={false}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 22,
                    "fontWeight": "bold",
                    "paddingVertical": 10,
                  }
                }
              >
                
              </Text>
            </View>
          </View>
          <ActivityIndicator
            animating={true}
            color="#999999"
            hidesWhenStopped={true}
            size="large"
            style={
              Object {
                "alignSelf": "center",
                "justifyContent": "center",
                "paddingTop": 70,
              }
            }
          />
          <View
            style={
              Object {
                "bottom": 0,
                "flexDirection": "row",
                "paddingBottom": 22,
                "paddingHorizontal": 22,
                "position": "absolute",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <View
                style={
                  Object {
                    "height": 10,
                  }
                }
              />
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                collapsable={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "opacity": 1,
                  }
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "rgb(230,37,101)",
                        "borderColor": "rgb(230,37,101)",
                        "borderRadius": 5,
                        "borderWidth": 1,
                        "paddingHorizontal": 10,
                        "paddingVertical": 5,
                      },
                      Object {
                        "backgroundColor": "#fff",
                        "borderColor": "transparent",
                      },
                    ]
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    disabled={false}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "color": "#FFF",
                          "fontSize": 17,
                        },
                        Object {
                          "color": "#808080",
                        },
                      ]
                    }
                  >
                    Cancel
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </Modal>
    <RCTScrollView>
      <View>
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                  Object {
                    "backgroundColor": "#F624591A",
                    "paddingBottom": 15,
                    "paddingTop": 0,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#F624598C",
                    "borderWidth": 0,
                    "padding": 10,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#fff",
                      "fontSize": 17,
                      "fontWeight": "400",
                    }
                  }
                >
                  Grocery List
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {},
                    Object {
                      "backgroundColor": "#FFF",
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": 0.5,
                      },
                      Object {
                        "backgroundColor": "#C8C7CC",
                        "marginLeft": 0,
                        "marginRight": 0,
                      },
                    ]
                  }
                />
              </View>
              <View
                accessibilityComponentType={undefined}
                accessibilityLabel={undefined}
                accessibilityTraits={undefined}
                accessible={true}
                hasTVPreferredFocus={undefined}
                hitSlop={undefined}
                isTVSelectable={true}
                nativeID={undefined}
                onLayout={undefined}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                    },
                    undefined,
                  ]
                }
                testID={undefined}
                tvParallaxProperties={undefined}
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "minHeight": 44,
                        "paddingLeft": 15,
                        "paddingRight": 20,
                      },
                      Object {
                        "backgroundColor": "#FFF",
                      },
                      Object {},
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "justifyContent": "center",
                        "marginRight": 15,
                      }
                    }
                  >
                    <Image
                      source={
                        Object {
                          "uri": "https://facebook.github.io/react/img/logo_og.png",
                        }
                      }
                      style={
                        Array [
                          Object {
                            "borderRadius": 5,
                          },
                          Object {
                            "height": 29,
                            "width": 29,
                          },
                        ]
                      }
                    />
                  </View>
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "paddingVertical": 10,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={false}
                      ellipsizeMode="tail"
                      numberOfLines={1}
                      style={
                        Array [
                          Object {
                            "flex": 1,
                            "fontSize": 16,
                            "letterSpacing": -0.32,
                          },
                          Object {
                            "color": "#000",
                          },
                          Object {},
                        ]
                      }
                    >
                      Meat
                    </Text>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={false}
                      ellipsizeMode="tail"
                      numberOfLines={1}
                      style={
                        Array [
                          Object {
                            "alignSelf": "center",
                            "color": "#8E8E93",
                            "fontSize": 16,
                            "letterSpacing": -0.32,
                          },
                          Object {
                            "color": "#8E8E93",
                          },
                          Object {},
                        ]
                      }
                    >
                       ~ 3.4 €
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "justifyContent": "center",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "backgroundColor": "transparent",
                          "borderColor": "#c7c7cc",
                          "borderRightWidth": 1,
                          "borderTopWidth": 1,
                          "height": 10,
                          "marginLeft": 7,
                          "transform": Array [
                            Object {
                              "rotate": "45deg",
                            },
                          ],
                          "width": 10,
                        }
                      }
                    />
                  </View>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {},
                    Object {
                      "backgroundColor": "#FFF",
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": 0.5,
                      },
                      Object {
                        "backgroundColor": "#C8C7CC",
                        "marginLeft": 0,
                        "marginRight": 0,
                      },
                    ]
                  }
                />
              </View>
              <View
                style={
                  Object {
                    "paddingLeft": 15,
                    "paddingRight": 15,
                    "paddingTop": 10,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "fontSize": 13,
                        "letterSpacing": -0.078,
                      },
                      Object {
                        "color": "#fff",
                      },
                    ]
                  }
                >
                  Total: 3.40 €
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "marginBottom": 10,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                  Object {
                    "backgroundColor": "#F624591A",
                    "paddingBottom": 0,
                    "paddingTop": 0,
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "backgroundColor": "#F624598C",
                    "borderWidth": 0,
                    "padding": 10,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  disabled={false}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#fff",
                      "fontSize": 17,
                      "fontWeight": "400",
                    }
                  }
                >
                  Details
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {},
                    Object {
                      "backgroundColor": "#FFF",
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": 0.5,
                      },
                      Object {
                        "backgroundColor": "#C8C7CC",
                        "marginLeft": 0,
                        "marginRight": 0,
                      },
                    ]
                  }
                />
              </View>
              <View>
                <View>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "minHeight": 44,
                          "paddingLeft": 15,
                          "paddingRight": 20,
                        },
                        Object {
                          "backgroundColor": "#FFF",
                        },
                        Object {},
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "flexDirection": "column",
                          "paddingVertical": 8,
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        disabled={false}
                        ellipsizeMode="tail"
                        style={
                          Object {
                            "color": "#353535",
                            "fontSize": 15,
                            "lineHeight": 25,
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={false}
                          disabled={false}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "#353535",
                                "fontSize": 14,
                              },
                              undefined,
                              Object {
                                "fontFamily": "Material Icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                            ]
                          }
                        >
                          
                        </Text>
                         
                        Location
                      </Text>
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        disabled={false}
                        ellipsizeMode="tail"
                        style={
                          Object {
                            "fontSize": 17,
                          }
                        }
                      >
                        Near undefined
                      </Text>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {},
                      Object {
                        "backgroundColor": "#FFF",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "height": 0.5,
                        },
                        Object {
                          "backgroundColor": "#C8C7CC",
                          "marginLeft": 15,
                          "marginRight": 0,
                        },
                      ]
                    }
                  />
                </View>
              </View>
              <View>
                <View>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "minHeight": 44,
                          "paddingLeft": 15,
                          "paddingRight": 20,
                        },
                        Object {
                          "backgroundColor": "#FFF",
                        },
                        Object {},
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "flexDirection": "column",
                          "paddingVertical": 8,
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        disabled={false}
                        ellipsizeMode="tail"
                        style={
                          Object {
                            "color": "#353535",
                            "fontSize": 15,
                            "lineHeight": 25,
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={false}
                          disabled={false}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "#353535",
                                "fontSize": 14,
                              },
                              undefined,
                              Object {
                                "fontFamily": "Material Icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                            ]
                          }
                        >
                          
                        </Text>
                         
                        Deliver by
                      </Text>
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        disabled={false}
                        ellipsizeMode="tail"
                        style={
                          Object {
                            "fontSize": 17,
                          }
                        }
                      >
                        10.05.17 12:15 am
                      </Text>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {},
                      Object {
                        "backgroundColor": "#FFF",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "height": 0.5,
                        },
                        Object {
                          "backgroundColor": "#C8C7CC",
                          "marginLeft": 15,
                          "marginRight": 0,
                        },
                      ]
                    }
                  />
                </View>
              </View>
              <View>
                <View>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "minHeight": 44,
                          "paddingLeft": 15,
                          "paddingRight": 20,
                        },
                        Object {
                          "backgroundColor": "#FFF",
                        },
                        Object {},
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "flexDirection": "column",
                          "paddingVertical": 8,
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        disabled={false}
                        ellipsizeMode="tail"
                        style={
                          Object {
                            "color": "#353535",
                            "fontSize": 15,
                            "lineHeight": 25,
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={false}
                          disabled={false}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "#353535",
                                "fontSize": 14,
                              },
                              undefined,
                              Object {
                                "fontFamily": "Material Icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                            ]
                          }
                        >
                          
                        </Text>
                         
                        Earn delivery fee
                      </Text>
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        disabled={false}
                        ellipsizeMode="tail"
                        style={
                          Object {
                            "fontSize": 17,
                          }
                        }
                      >
                        12 €
                      </Text>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {},
                      Object {
                        "backgroundColor": "#FFF",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "height": 0.5,
                        },
                        Object {
                          "backgroundColor": "#C8C7CC",
                          "marginLeft": 15,
                          "marginRight": 0,
                        },
                      ]
                    }
                  />
                </View>
              </View>
              <View>
                <View>
                  <View
                    style={
                      Array [
                        Object {
                          "alignItems": "center",
                          "flexDirection": "row",
                          "minHeight": 44,
                          "paddingLeft": 15,
                          "paddingRight": 20,
                        },
                        Object {
                          "backgroundColor": "#FFF",
                        },
                        Object {},
                      ]
                    }
                  >
                    <View
                      style={
                        Object {
                          "flexDirection": "column",
                          "paddingVertical": 8,
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        disabled={false}
                        ellipsizeMode="tail"
                        style={
                          Object {
                            "color": "#353535",
                            "fontSize": 15,
                            "lineHeight": 25,
                          }
                        }
                      >
                        <Text
                          accessible={true}
                          allowFontScaling={false}
                          disabled={false}
                          ellipsizeMode="tail"
                          style={
                            Array [
                              Object {
                                "color": "#353535",
                                "fontSize": 14,
                              },
                              undefined,
                              Object {
                                "fontFamily": "Material Icons",
                                "fontStyle": "normal",
                                "fontWeight": "normal",
                              },
                            ]
                          }
                        >
                          
                        </Text>
                         
                        Status
                      </Text>
                      <Text
                        accessible={true}
                        allowFontScaling={true}
                        disabled={false}
                        ellipsizeMode="tail"
                        style={
                          Object {
                            "fontSize": 17,
                          }
                        }
                      >
                        Pending
                      </Text>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    Array [
                      Object {},
                      Object {
                        "backgroundColor": "#FFF",
                      },
                    ]
                  }
                >
                  <View
                    style={
                      Array [
                        Object {
                          "height": 0.5,
                        },
                        Object {
                          "backgroundColor": "#C8C7CC",
                          "marginLeft": 15,
                          "marginRight": 0,
                        },
                      ]
                    }
                  />
                </View>
              </View>
              <View>
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "minHeight": 44,
                        "paddingLeft": 15,
                        "paddingRight": 20,
                      },
                      Object {
                        "backgroundColor": "#FFF",
                      },
                      Object {},
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flexDirection": "column",
                        "paddingVertical": 8,
                      }
                    }
                  >
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "color": "#353535",
                          "fontSize": 15,
                          "lineHeight": 25,
                        }
                      }
                    >
                      <Text
                        accessible={true}
                        allowFontScaling={false}
                        disabled={false}
                        ellipsizeMode="tail"
                        style={
                          Array [
                            Object {
                              "color": "#353535",
                              "fontSize": 14,
                            },
                            undefined,
                            Object {
                              "fontFamily": "Material Icons",
                              "fontStyle": "normal",
                              "fontWeight": "normal",
                            },
                          ]
                        }
                      >
                        
                      </Text>
                       
                      Purchased at
                    </Text>
                    <Text
                      accessible={true}
                      allowFontScaling={true}
                      disabled={false}
                      ellipsizeMode="tail"
                      style={
                        Object {
                          "fontSize": 17,
                        }
                      }
                    >
                      10.05.17 12:15 am
                    </Text>
                  </View>
                </View>
              </View>
              <View
                style={
                  Array [
                    Object {},
                    Object {
                      "backgroundColor": "#FFF",
                    },
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": 0.5,
                      },
                      Object {
                        "backgroundColor": "#C8C7CC",
                        "marginLeft": 0,
                        "marginRight": 0,
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
    <View
      style={
        Object {
          "flexDirection": "row",
          "paddingHorizontal": 22,
          "paddingVertical": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "rgb(230,37,101)",
                  "borderColor": "rgb(230,37,101)",
                  "borderRadius": 5,
                  "borderWidth": 1,
                  "paddingHorizontal": 10,
                  "paddingVertical": 10,
                },
                Object {
                  "backgroundColor": "#c1c1c1",
                  "borderColor": "transparent",
                },
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              disabled={false}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "#FFF",
                  "fontSize": 18,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
